kernel/fablib/runlevel/0_infrastructure/aws_ssh_key/aws_ssh_key.go

```diff
@@ -0,0 +1,120 @@
+package aws_ssh_key
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/openziti/fablab/kernel/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"io/ioutil"
+	"path"
+)
+
+var KeyManager = awsKeyManager{}
+
+func Express() model.InfrastructureStage {
+	return KeyManager
+}
+
+type awsKeyManager struct {
+}
+
+func (l awsKeyManager) Bootstrap(m *model.Model) error {
+	bindings := model.GetBindings()
+
+	if !bindings.Has("credentials", "aws", "ssh_key_name") {
+		bindings.Put(bindings.Must("environment"), "credentials", "aws", "ssh_key_name")
+	}
+
+	if bindings.Has("credentials", "ssh", "key_path") {
+		return nil
+	}
+
+	keyPath := path.Join(model.ActiveInstancePath(), "ssh_private_key.pem")
+
+	bindings.Put(keyPath, "credentials", "ssh", "key_path")
+	return nil
+}
+
+func (stage awsKeyManager) Express(m *model.Model, l *model.Label) error {
+	bindings := model.GetBindings()
```

Comment:
Here's where it check for the credentials.aws.managed_key flag. If not set to true, no key management is done for you